pkg/util/fuzzystrmatch/metaphone.go

```diff
@@ -0,0 +1,347 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package fuzzystrmatch
+
+import (
+	"strings"
+	"unicode"
+)
+
+// Binstock, Andrew & Rex, John. "Metaphone: A Modern Soundex." Practical Algorithms For Programmers.
+// Reading, Mass: Addion-Wesley, 1995 pp160-169
+
+// Special encodings
+const SH = 'X'
+const TH = '0'
+
+/* Port from Postgres */
+var _codes = [26]int{1, 16, 4, 16, 9, 2, 4, 16, 9, 2, 0, 2, 2, 2, 1, 4, 0, 2, 4, 4, 1, 0, 0, 0, 8, 0}
+
+//					 a  b   c  d   e  f  g  h   i  j  k  l  m  n  o  p  q  r  s  t  u  v  w  x  y  z
```

Comment:
nit: make sure the whitespace alignment doesn't use tabs here.

Reply:
done!